### ISSUE TITLE
dnvm install creates empty folders under ~/.dnx/runtimes if the version is not available

### ISSUE THREAD

**Issue by sayedihashimi** [aspnet/dnvm#327]
Try this on Mac ```dnvm install 1.0.0-beta5``` The result is

```
Sayeds-MacBook-Pro:WebApplication_IndividualAuth sayedhashimi$ dnvm install 1.0.0-beta5
Downloading dnx-mono.1.0.0-beta5 from https://www.nuget.org/api/v2
Download: https://www.nuget.org/api/v2/package/dnx-mono/1.0.0-beta5
######################################################################## 100.0%
dnx-mono.1.0.0-beta5 was not found in repository https://www.nuget.org/api/v2
This is most likely caused by the feed not having the version that you typed. Check that you typed the right version and try again. Other possible causes are the feed doesn't have a DNX of the right name format or some other error caused a 404 on the server.
```

Then try ```dnvm list```

```
Active Version              Runtime Arch Location             Alias
------ -------              ------- ---- --------             -----
  *    1.0.0-beta4          mono         ~/.dnx/runtimes      default
       1.0.0-beta5          mono         ~/.dnx/runtimes
       1.0.0-beta6-12100    mono         ~/.dnx/runtimes
```

So it looks installed, let's see what the folder contains.

```
ls -l -a  ~/.dnx/runtimes/dnx-mono.1.0.0-beta5
total 16
drwxr-xr-x  3 sayedhashimi  staff   102 Jun 19 15:32 .
drwxr-xr-x  5 sayedhashimi  staff   170 Jun 19 15:32 ..
-rw-r--r--  1 sayedhashimi  staff  7835 Jun 19 15:32 dnx-mono.1.0.0-beta5.nupkg
```

As you can see it just contains a .nupkg file that is 7.7k and none of the actual content.

Let's see what happens when we try to install again with ```dnvm install 1.0.0-beta5```

```
dnx-mono.1.0.0-beta5 already installed.
Cannot find dnx-mono.1.0.0-beta5, do you need to run 'dnvm install 1.0.0-beta5'?
```

So it's installed but needs to be installed.

This is a really bad user experience. I know that we saw this with ```dnvm``` on Windows and fixed it a while back. When are we fixing this for xplat?

**Comment by gingters** [comment]
+1 for that. It really should not create the folder before the runtime has been found and is started being downloaded, or, when download was unsuccessful, remove the folder afterwards.

**Comment by sayedihashimi** [comment]
Any update here? I was just about to open a new issue on this but then relalized I already opened this one.

On Mac when you call `dnvm install` with a name that doesn't exist, for example `dnvm install 1.0.0-beta7` when using nuget.org, the result is that in `~/.dnx/runtimes` you have a `1.0.0-beta7` folder which is ***empty***.

After running the `dnvm install` command when you execute `dnvm list` it ***looks like*** the runtime is installed but it's not. Here is the output that I'm seeing.

```
Sayeds-MacBook-Pro:WebApplication sayedhashimi$ dnvm install 1.0.0-beta7
Downloading dnx-mono.1.0.0-beta7 from https://www.nuget.org/api/v2
Download: https://www.nuget.org/api/v2/package/dnx-mono/1.0.0-beta7
######################################################################## 100.0%
Installing to /Users/sayedhashimi/.dnx/runtimes/dnx-mono.1.0.0-beta7
find: /Users/sayedhashimi/.dnx/runtimes/dnx-mono.1.0.0-beta7/bin/: No such file or directory
chmod: /Users/sayedhashimi/.dnx/runtimes/dnx-mono.1.0.0-beta7/bin/dnx: No such file or directory
Sayeds-MacBook-Pro:WebApplication sayedhashimi$ dnvm list

Active Version              Runtime Arch Location             Alias
------ -------              ------- ---- --------             -----
       1.0.0-beta5          mono         ~/.dnx/runtimes
  *    1.0.0-beta6          mono         ~/.dnx/runtimes
       1.0.0-beta7          mono         ~/.dnx/runtimes
       1.0.0-beta7-12264    mono         ~/.dnx/runtimes      default

Sayeds-MacBook-Pro:WebApplication sayedhashimi$ ls ~/.dnx/runtimes/dnx-mono.1.0.0-beta7
Sayeds-MacBook-Pro:WebApplication sayedhashimi$

Sayeds-MacBook-Pro:WebApplication sayedhashimi$ dnvm use 1.0.0-beta7
Cannot find dnx-mono.1.0.0-beta7, do you need to run 'dnvm install 1.0.0-beta7'?
```

This is especially problematic because dnvm doesn't have an [uninstall command](https://github.com/aspnet/dnvm/issues/254) so users have to know to delete those empty folders.

I'd really like to get this fixed.

**Comment by mholo65** [comment]
@anurse I could fix this during the weekend and add also some tests. 

Fix should be straightforward, in **__dnvm_download()** remove previously created **$runtimeFolder** if **$httpResult** is **404**... Or anything else than **200** or **302**

[src/dnvm.sh](https://github.com/aspnet/dnvm/blob/f13c8521b5c08c2f343b7aca4978f78697bc1261/src/dnvm.sh#L241-L246)

**Comment by mholo65** [comment]
There's also another issue. we are looking at the first HTTP-result (***grep "^HTTP/1.1" | head -n 1***), which will almost always be a redirect (302). The last HTTP-result would give the actual result **200** or **404**. We should therefore instead check the last HTTP-result (***grep "^HTTP/1.1" | tail -n 1***).

This is why @sayedihashimi said that it ***looks like*** the runtime is installed.

Here's an output from curl which proves the above:
```
curl -L -D - "https://www.nuget.org/api/v2/package/dnx-mono/1.0.0-beta7" -o "1.0.0-beta7.nupkg" -# -k

HTTP/1.1 302 Found
Cache-Control: private
Content-Type: text/html; charset=utf-8
Location: https://az320820.vo.msecnd.net/packages/dnx-mono.1.0.0-beta7.nupkg
Server: Microsoft-IIS/8.5
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-Content-Type-Options: nosniff
Strict-Transport-Security: max-age=31536000
Date: Sat, 01 Aug 2015 10:20:15 GMT
Content-Length: 183

######################################################################## 100,0%
HTTP/1.1 404 Not Found
Content-Type: application/xml
Date: Sat, 01 Aug 2015 10:20:15 GMT
Server: Windows-Azure-Blob/1.0 Microsoft-HTTPAPI/2.0
x-ms-request-id: 1b8a263d-0001-0005-7343-cc7ddb000000
x-ms-version: 2009-09-19
Content-Length: 215

######################################################################## 100,0%
```

**Comment by mholo65** [comment]
Hmm, can't seem to reproduce anymore. Was maybe fixed with this 1aaa9fa

However, the original 404 error message (and check) in __dnvm_download() is not used anymore. Should maybe be moved?

**Comment by glennc** [comment]
Closing as duplicate of https://github.com/aspnet/dnvm/issues/189
